wordpress-mobile/WordPress-iOS
# Jetpack Social: Add Core Data model for PublicizeInfo

Refs #20806

> **Note**:  This PR targets 22.8 in case there's not enough time to merge this before the 22.7 branch is cut on Monday.

As titled, this adds the Core Data model for `PublicizeInfo`, which is a direct mapping of `RemotePublicizeInfo`. I'm intentionally isolating this PR since it contains quite a lot of changes.

## To test

There's nothing to test as of now, but please verify if there are any configurations that I've missed:

- The current model version should be set to 151.
- The one-to-one relationship between `Blog` and `PublicizeInfo` is optional for both ways.
- All the properties of `PublicizeInfo` are not optional, but the value defaults to `0`.
- The property mappings in `PublicizeInfo+CoreDataProperties` matches the model definition in the xcdatamodel.
- Check the additions in `MIGRATIONS.md` and verify that all the changes are covered and correct.

## Regression Notes
1. Potential unintended areas of impact
Should be none. Model is unused, and feature is hidden behind a flag.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A.

3. What automated tests I added (or what prevented me from doing so)
N/A.

## PR submission checklist

N/A. This PR doesn't contain any UI changes.


UI Changes testing checklist:
